From ba32d3525584b6af33f1999b5dcbb602b5bdc691 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Mon, 25 Feb 2019 16:02:11 +0100
Subject: [PATCH 0272/1239] pinctrl: mvebu: allow proper SD/MMC PHY MUX setting
 in CP115

SD/MMC PHY usage requires additional setting in MPP registers.
For standalone CP115 chips it wasn't done, due to wrong compatible
strings for the pinctrl. Update it to be on par with Linux DT and
modify the MPP driver accordingly.

Change-Id: I2e566ae9be9b1f6e2ea3a45d80a04ca0c70e0b4e
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
---
 drivers/pinctrl/mvebu/pinctrl-mvebu.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/mvebu/pinctrl-mvebu.c b/drivers/pinctrl/mvebu/pinctrl-mvebu.c
index e18b39dc58..e8423d550b 100644
--- a/drivers/pinctrl/mvebu/pinctrl-mvebu.c
+++ b/drivers/pinctrl/mvebu/pinctrl-mvebu.c
@@ -51,7 +51,9 @@ void mvebu_pinctl_emmc_set_mux(struct udevice *dev, u32 pin, u32 func)
 	} else if (!fdt_node_check_compatible(blob, node,
 					"marvell,armada-8k-cpm-pinctrl") ||
 		   !fdt_node_check_compatible(blob, node,
-					"marvell,armada-7k-pinctrl")) {
+					"marvell,armada-7k-pinctrl") ||
+		   !fdt_node_check_compatible(blob, node,
+					"marvell,cp115-standalone-pinctrl")) {
 		if ((pin == CP110_EMMC_CLK_PIN_ID) &&
 		    (func == CP110_EMMC_CLK_FUNC)) {
 			clrbits_le32(priv->base_reg + CP_EMMC_PHY_CTRL_REG,
@@ -222,6 +224,7 @@ static const struct udevice_id mvebu_pinctrl_ids[] = {
 	{ .compatible = "marvell,armada-7k-pinctrl" },
 	{ .compatible = "marvell,armada-8k-cpm-pinctrl" },
 	{ .compatible = "marvell,armada-8k-cps-pinctrl" },
+	{ .compatible = "marvell,cp115-standalone-pinctrl" },
 	{ }
 };
 
-- 
2.29.0

